api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto

```diff
@@ -174,4 +176,16 @@ message HttpProtocolOptions {
   // [#not-implemented-hide:]
   // [#extension-category: envoy.http.header_validators]
   config.core.v3.TypedExtensionConfig header_validation_config = 7;
+
+  message HttpEvents {
+    // Matcher for response headers.
+    config.common.matcher.v3.MatchPredicate match = 1 [(validate.rules).message = {required: true}];
+
+    // Code which should be reported to the outlier detection if response headers matched the matcher.
+    envoy.type.v3.StatusCode map_to = 2 [(validate.rules).enum = {defined_only: true}];
+  }
+
+  // List of matchers for response headers along with codes to be reported to outlier detection.
+  // The first matcher which returns matching success is applied.
+  repeated HttpEvents outlier_detection = 8;
```

Comment:
Why we need a list of `HttpEvents`? I think single match should enough to match an error?

```
message OutlierDetection {
  config.common.matcher.v3.MatchPredicate failure_matcher = 1 [(validate.rules).message = {required: true}];
}

OutlierDetection outlier_detection = 8;
```